apps/examples/lcd_test/example_lcd.c

```diff
@@ -254,13 +254,105 @@ static void test_bit_map(void)
 	free(lcd_data);
 }
 
+static void test_fps(void)
+{
+	int fd_rtc = 0;
+	int fd_lcd = 0;
+	int p = 0;
+	char port[20] = { '\0' };
+	size_t len;
+
+	fd_rtc = open("/dev/rtc0", O_RDWR);
+	if (fd_rtc < 0) {
+		printf("ERROR: LCD FPS test, Fail to open rtc.\n");
+		return;
+	}
+
+	len = xres * yres * 2 + 1;
+	uint8_t *lcd_data_red = (uint8_t *)malloc(len);
+	uint8_t *lcd_data_blue = (uint8_t *)malloc(len);
+	if (lcd_data_red == NULL || lcd_data_blue == NULL) {
+		printf("FPS TEST, malloc failed for lcd data : %d\n", len);
+		close(fd_rtc);
+		return;
+	}
+	for (int i = 0; i < len-1; i += 2) {
+		lcd_data_red[i] = (RED & 0xFF00) >> 8;
+		lcd_data_red[i + 1] = RED & 0x00FF;
+		lcd_data_blue[i] = (BLUE & 0xFF00) >> 8;
+		lcd_data_blue[i + 1] = BLUE & 0x00FF;
+	}
+
+	struct lcddev_area_s area_red;
+	struct lcddev_area_s area_blue;
+	area_red.planeno = 0;
+	area_red.row_start = 0;
+	area_red.row_end = yres - 1;
+	area_red.col_start = 0;
+	area_red.col_end = xres - 1;
+	area_red.stride = 2 * xres;
+	area_red.data = lcd_data_red;
+	area_blue.planeno = 0;
+	area_blue.row_start = 0;
+	area_blue.row_end = yres - 1;
+	area_blue.col_start = 0;
+	area_blue.col_end = xres - 1;
+	area_blue.stride = 2 * xres;
+	area_blue.data = lcd_data_blue;
+
+	sprintf(port, LCD_DEV_PATH, p);
+	fd_lcd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd_lcd < 0) {
+		printf("ERROR: FPS TEST, Failed to open lcd port : %s error:%d\n", port, fd_lcd);
+		free(lcd_data_red);
+		free(lcd_data_blue);
+		close(fd_rtc);
+		return;
+	}
+
+	struct rtc_time start_time = RTC_TIME_INITIALIZER(1970, 1, 1, 0, 0, 0);
+	struct rtc_time end_time;
+
+	bool is_red = true;
+	//Start Test
+	ioctl(fd_rtc, RTC_RD_TIME, (unsigned long)&start_time);
+	for (int itr = 0; itr < FPS_TEST_FRAME_COUNT; itr++) {
+		if (is_red) {
+			ioctl(fd_lcd, LCDDEVIO_PUTAREA, (unsigned long)(uintptr_t)&area_red);
+			is_red = false;
+		} else {
+			ioctl(fd_lcd, LCDDEVIO_PUTAREA, (unsigned long)(uintptr_t)&area_blue);
+			is_red = true;
+		}
+	}
+	ioctl(fd_rtc, RTC_RD_TIME, (unsigned long)&end_time);
+	//End test
+	
+	close(fd_rtc);
+	close(fd_lcd);
+	free(lcd_data_red);
+	free(lcd_data_blue);
+
+	time_t start;
+	time_t end;
+	start = mktime((FAR struct tm *)&start_time);
+	end = mktime((FAR struct tm *)&end_time);
+
+	if (difftime(end, start) != 0) {
+		float fps = FPS_TEST_FRAME_COUNT / difftime(end, start);
+		printf("FPS: %.2f\n", fps);
+	} else {
+		printf("FPS calculation failed! Please check start/end timediff!\n");
```

Comment:
> 1 frame is taking like 16 ms. So if the number of frames executed is less than 62, then the start and end will be same because rtc records time in seconds.

I want to print `failed, so how to resolve`. When we meet this case, you can increase `FPS_TEST_FRAME_COUNT` to resolve the failure. If right, let's explain this. But this will requires config for FPS_TEST_FRAME_COUNT definition.

Reply:
ok..  so,   how about if we make a config "CONFIG_EXAMPLE_LCD_FPS_TEST". If this value is not defined, then we won't execute this test, otherwise this test case will be executed with the value defined.
CONFIG_EXAMPLE_LCD_FPS_TEST=1000